WooCommerce/src/main/kotlin/com/woocommerce/android/ui/payments/feedback/ipp/GetIPPFeedbackBannerData.kt

```diff
@@ -39,58 +43,52 @@ class GetIPPFeedbackBannerData @Inject constructor(
             return IPP_NEWBIE_BANNER
         }
 
-        val timeWindowStartDate = Date().daysAgo(STATS_TIME_WINDOW_LENGTH_DAYS)
-        val response = ippStore.fetchTransactionsSummary(
-            activePaymentsPlugin,
-            siteModel,
-            timeWindowStartDate.formatToYYYYmmDD()
-        )
+        val wcPayIPPOrders = getIppOrders()
+        val hasWCPayIPPOrders = wcPayIPPOrders.isNotEmpty()
+        val wcPayIPPOrdersCount = wcPayIPPOrders.size
 
-        if (!response.isSuccessful()) {
-            logger.e(AppLog.T.API, "Error fetching transactions summary: ${response.error}")
-            return IPP_NEWBIE_BANNER
-        }
-
-        val numberOfTransactionsInLast30Days = response.result?.transactionsCount
-        if (numberOfTransactionsInLast30Days == null || numberOfTransactionsInLast30Days < 0) {
-            logger.e(AppLog.T.API, "Transactions count data is malformed.")
-            return null
-        }
+        val wcPayIPPOrdersInLast30Days = getIppOrdersforLast(STATS_TIME_WINDOW_LENGTH_DAYS)
+        val hasIPPOrdersInLast30Days = wcPayIPPOrdersInLast30Days.isNotEmpty()
+        val ippOrdersInLast30Days = wcPayIPPOrdersInLast30Days.size
 
-        return when (numberOfTransactionsInLast30Days) {
-            0 -> if (hasUserEverMadeIppTransaction()) {
-                IPP_BEGINNER_BANNER
-            } else {
+        return when {
+            !hasWCPayIPPOrders && cashOnDeliverySettings.isCashOnDeliveryEnabled() -> {
                 IPP_NEWBIE_BANNER
             }
-            in IPP_BEGINNER_TRANSACTIONS_RANGE -> IPP_BEGINNER_BANNER
-            else -> IPP_NINJA_BANNER
+            hasIPPOrdersInLast30Days && ippOrdersInLast30Days in IPP_BEGINNER_TRANSACTIONS_RANGE -> {
+                IPP_BEGINNER_BANNER
+            }
+            wcPayIPPOrdersCount >= IPP_NINJA_TRANSACTIONS_THRESHOLD -> {
```

Comment:
```suggestion
            wcPayIPPOrdersCount > IPP_BEGINNER_TRANSACTIONS_RANGE.last -> {
```
🔍 `IPP_NINJA_TRANSACTIONS_THRESHOLD` seems to be not necessary - we can refer to `IPP_BEGINNER_TRANSACTIONS_RANGE.last`.

Reply:
Added here: https://github.com/woocommerce/woocommerce-android/pull/8482/commits/b610dc23add12acd801ef2727185d28d326691eb